src/transaction/transaction.service.ts

```diff
@@ -1,7 +1,34 @@
 import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Transaction } from 'src/entities/transaction.entity';
+
 
 @Injectable()
 export class TransactionService {
+  @InjectRepository(Transaction)
+  private readonly transactionRepository: Repository<Transaction>
+
+
+
+  async saveTransaction(idMarchand: string, amount: number){
+
+    try{
+     const transaction = new Transaction();
+    //  transaction.idMarchand = idMarchand;
+     transaction.amount = amount;
+     const saveTransaction : await this.transactionRepository.save(transaction);
```

Comment:
c'est une constante qu'on declare donc on utilise le symbole = 

=> 
`const saveTransaction = await this.transactionRepository.save(transaction);`